### ISSUE TITLE
Ensure consistent Harp device core version and firmware version

### ISSUE THREAD

**Issue by alexpiet** [AllenNeuralDynamics/dynamic-foraging-task#552]
We should:
1. programmatically log the firmware version of each device for Metadata and the gui logs
2. ensure that all rigs are using the same firmware version on each device type (soundcard, behavior board, etc)
3. Ensure that new boards from SIPE come with that firmware version

**Comment by alexpiet** [comment]
@hagikent Says he updated 446-6/7 to soundcard firmware 2.2

**Comment by hagikent** [comment]
>2. ensure that all rigs are using the same firmware version on each device type (soundcard, behavior board, etc)
proper fw would be different depending on hardwae versions.
at least for soundcard, several hw versions (2.1 / 2.2) are intermingled in 446/447.

>3. Ensure that new boards from SIPE come with that firmware version. Right now SIPE just gives us the board with whatever firmware was on the board when it arrived at the institute
Is Nathan in charge of this too? If so, I can transfer how-to to him.

**Comment by ellahiltonvano** [comment]
@hagikent @alexpiet does it make sense to ask SIPE to install the correct firmware for the upcoming upgrades to 447?

**Comment by alexpiet** [comment]
Yes

**Comment by ellahiltonvano** [comment]
@hagikent I've added a comment to the 447 upgrade ticket to see if this is within the scope of SIPE. I'll update when I hear back but if so it would be great if you could transfer this knowledge

**Comment by hagikent** [comment]
Yup, should be easy for them (at least behavior board). Sound card is tricky because of an inherent issue of the product. I can directly connect with Nathan or other SIPE members. That said, should be fine as is if they keep using the exact same HARP boards (i.e. no replacement).

**Comment by ellahiltonvano** [comment]
@hagikent Nathan commented this on the ticket: 
"We're planning to reuse the boards (except new motor controller) so the firmware should be correct. Kenta can certainly send us any info we should have going forward, though."

Does it make sense to have you add any relevant information on this ticket? Let me know how I can help. http://mpe-redirects/MPETracking/ItemID=3616

**Comment by hagikent** [comment]
ok, I will stub corresponding instruction to the ticket.

**Comment by ellahiltonvano** [comment]
@alexpiet should this issue be kept open since Kenta is providing instructions to SIPE for future setups? There seem to be a couple other items, not sure if they have already been addressed

**Comment by alexpiet** [comment]
lets keep it open

**Comment by alexpiet** [comment]
related to #1435

**Comment by ellahiltonvano** [comment]
SIPE ticket: https://apps.powerapps.com/play/cb338490-e23d-48e6-9e6d-e600b8d17dd4?tenantId=32669cd6-737f-4b39-8bdd-d6951120d3fc&Hidenavbar=true&ItemID=4304
